architecture/README.md

```diff
@@ -1 +1,147 @@
+**Cloudkit Controller**: A Kubernetes operator running on each management cluster that watches for orders and then ensures they get fulfilled by using a combination of direct automation and delegation to Event Driven Ansible.
+**Ansible Automation Platform (AAP)**: Executes the majority of provisioning steps by running the associated Templates.
+
+The general workflow for fulfillment is as follows:
+
+* First, the cloud provider makes a request for resources through the API, on behalf of a tenant user, having received that request through their own user interface. Alternatively the cloud provider might expose the fulfillment API directly to end users.
+* The fulfillment service selects a management cluster and then places the order there with the Kubernetes CloudKit operator using Kubernetes APIs.
+* The CloudKit operator then performs automated setup and triggers the third component: Ansible playbooks via Event-Driven Ansible (EDA). It then reports status back to the Fulfillment Service.
+* Once the EDA webhook is triggered, AAP runs to manage the deployment. The exact automation varies and can be customized by each cloud provider’s needs.
+
+The Fulfillment Service is intentionally modeled on the Kubernetes pattern:
+providers submit an Order that declares the desired state, and the CloudKit
+operator reconciles the cluster to match that desired state. To put it simply, the
+fulfillment API creates the order, and the Cloudkit Operator reconciles the
+order. This gives the service a flexible way to achieve reconciliation that
+doesn’t limit us to a specific set of APIs or tools. Because this offers a wider
+application, further configuration to fit the needs of service providers is done
+through Ansible Automation Platform (AAP).
+
+The flexibility of this architecture means that, broadly speaking, there are
+only two things that must happen when implementing a fulfillment workflow.
+
+1. Define an order object that has all the attributes a user needs to specify their desired outcome. For cluster fulfillment, that's simply a template and template parameters.
+2. Create the playbooks that reconcile the order. For cluster fulfillment, those playbooks call the OpenShift provisioning APIs, as well as the beginnings of bare metal/L2/L3 workflows.
+
+Details of specific fulfillment types are specified in the following sections.
+
+* [Cluster Fulfillment](cluster-fulfillment.md)
+* [Bare Metal Server Fulfillment](bm-server-fulfillment.md)
+* [VM Fulfillment](vm-fulfillment.md)
+
+### Integration Patterns
+
+In many cases, a CSP will use its own APIs and services to interact with
+end-users, and then the CSP's systems will call the Fulfillment Service in
+response to end-user actions. In those scenarios, the end-user never interacts
+directly with the Fulfillment Service.
+
```

Comment:
Is it worth calling out that we expect this to be the most common model of interacting with the solution?